AprilSylph/XKit-Rewritten
# Popup: Clear storage optionally on backup restore

### Description


This adds a checkbox to the backup restore interface in the specific scenario where the user's pasted backup is missing some of the values that the extension storage currently contains. As noted in the linked issue, this scenario currently results in those old values persisting. With this checkbox selected, the extension will remove those values as part of the restore process.

Most people should not see this, as they're presumably loading a backup with the same amount of information or more information than their storage currently contains

Resolves #1191.

### Testing steps

- Export your data, if you're testing this with your real data. Toggle some preferences to generate some data if you aren't.

<br />

- Paste your current data into the restore box. Confirm that no checkbox appears and that restoring works as expected.

<br />

- Paste `{}` into the restore box. Uncheck the resulting checkbox. Confirm that restoring works as expected but results in no change.
- Paste a subset of your current data into the restore box with a value change. Uncheck the resulting checkbox. Confirm that restoring works as expected but results in no change.
- Paste `{}` into the restore box. Check the resulting checkbox. Confirm that restoring removes all of your data.
